raiden/tests/integration/network/transport/rif_comms/node.py

```diff
@@ -0,0 +1,67 @@
+from subprocess import Popen
+
+import json5
+
+from raiden.tests.integration.network.transport.rif_comms.process import RIF_COMMS_PATH, Process as CommsProcess
+from raiden.tests.integration.network.transport.utils import generate_address
+from transport.rif_comms.client import Client as RIFCommsClient
+
+
+class Config:
+    """
+    Class to load and set configuration attributes for a RIF Comms node to be run.
+    """
+
+    api_endpoint_prefix = "localhost"
+    env_file_prefix = "testing_"
+
+    def __init__(self, node_number: int):
+        """
+        Load and set a configuration attributes for a RIF Comms node.
+        A valid configuration file must exist at RIF_COMMS_PATH/config/testing_<node_number>.json5
+        """
+        self.env_name = self.env_file_prefix + str(node_number)
+        # TODO: generate these files if needed
+        self._env_file = RIF_COMMS_PATH.joinpath('config/' + self.env_name + '.json5')
+
+        # load config from file
+        with open(self._env_file, 'r') as reader:
+            config = json5.loads(reader.read())
+            self._listening_port = config['grpcPort']
+
+        self.address = generate_address()
+        self.api_endpoint = self.api_endpoint_prefix + ":" + str(self._listening_port)
+
+
+class Node:
+    """
+    Class for RIF Comms node program.
+    """
+    connections = {}  # hack to get around the fact that each connect() call needs to be assigned
+
+    def __init__(self, config: Config):
+        self.address = config.address
+        self._api_endpoint = config.api_endpoint
+        self._env_name = config.env_name
+
+        self.client = RIFCommsClient(rsk_address=self.address, grpc_api_endpoint=self._api_endpoint)
+        self._process = self.start()
+
+    def start(self) -> Popen:
+        """
+        Start a RIF Comms node process and connect to it.
+        """
+        process = CommsProcess.start(env_name=self._env_name)
+        # FIXME: client.connect() calls should not need assignment
+        self.connections[self.address] = self.client.connect()
+        return process
+
+    def stop(self):
+        """
+        Disconnect from RIF Comms node program, and stop its process.
+        """
+        try:
+            # FIXME: deleting entries in the connections dictionary is causing non-crashing thread exceptions
```

Comment:
> does this implies any incorrect behaviour on the tests?
> 
> or just throws some exceptions that doesnt affect the test resutls?

i've seen some exceptions pop up in the case of deleting these entries during the execution of the teardown code, and they look like this:
> python grpcio 
> RuntimeError: cannot exit context: thread state references a different context object

since this occurs _after_ the tests are executed, it does not affect their result (be it success or fail). 

having said this, i would still rather not have these exceptions, so this is why i did not touch the dictionary at this point.

you can take a look at this for yourself by adding `del self.connections[self.address]` on line `65` and then running `pytest -s`.